src/transport/manager/mod.rs

```diff
@@ -711,6 +750,23 @@ impl TransportManager {
                 record,
                 dial_record: Some(mut dial_record),
             } => {
+                if dial_record.connection_id() != &Some(connection_id) {
```

Comment:
The same as above, but when the dial attempt using the first known address succeeded?